src/server/repositories/UrlRepository.ts

```diff
@@ -0,0 +1,240 @@
+import { inject, injectable } from 'inversify'
+import { BaseRepository } from './BaseRepository'
+import { Url, UrlType } from '../models/url'
+import { NotFoundError } from '../util/error'
+import { redirectClient } from '../redis'
+import { logger, redirectExpiry } from '../config'
+import { transaction } from '../util/sequelize'
+import DependencyIds from '../constants'
+import { FileVisibility, S3Interface } from '../util/aws'
+import { ACTIVE } from '../models/types'
+
+const { Public, Private } = FileVisibility
+
+/**
+ * A url repository that handles access to the data store of Urls.
+ */
+@injectable()
+export abstract class AbstractUrlRepository
```

Comment:
may i ask what drove your thinking to use an abstract class instead of an interface?

Reply:
I just realised this and committed the change. Was initially experimenting with sharing methods between repositories but abandoned it for the reason given in the pr